pkg/client/cache/fifo.go

```diff
@@ -183,12 +186,12 @@ func (f *FIFO) GetByKey(key string) (item interface{}, exists bool, err error) {
 	return item, exists, nil
 }
 
-// Pop waits until an item is ready and returns it. If multiple items are
+// Pop waits until an item is ready and processes it. If multiple items are
 // ready, they are returned in the order in which they were added/updated.
-// The item is removed from the queue (and the store) before it is returned,
-// so if you don't successfully process it, you need to add it back with
+// The item is removed from the queue (and the store) before it is processed,
+// so if you don't successfully process it, it should be added back with
```

Comment:
Add comment about how the process semantics work (i.e. anything passed to popprocessfunc is done under the lock and is safe, etc)